gnovm/pkg/gnolang/values.go

```diff
@@ -70,7 +72,50 @@ var (
 // ----------------------------------------
 // StringValue
 
-type StringValue string
+type StringValue struct {
+	isNewBase bool // if the underlying data is new allocated or not.
+	Base      *SliceValue
+}
```

Comment:
see alloc_7.gno.

Reply:
Is it worthy? The memory usage of the string value has obviously increased.

Reply:
No let's revert to the old StringValue string. 
Go does optimizations already, and a primitive value should not hold an object.
I suspect we can use unsafe to get any string's pointer, and use an optimized map to cache things,
but for now let's just assume every string is separate.

Reply:
ok thanks.

Reply:
fixed.